server/src/services/lbajob.service.ts

```diff
@@ -0,0 +1,265 @@
+import { encryptMailWithIV } from "../common/utils/encryptString.js"
+import { IApiError, manageApiError } from "../common/utils/errorManager.js"
+import { trackApiCall } from "../common/utils/sendTrackingEvent.js"
+import { filterJobsByOpco } from "./opco.service.js"
+
+const coordinatesOfFrance = [2.213749, 46.227638]
+
+import { NIVEAUX_POUR_LBA } from "../common/constants.js"
+import { roundDistance } from "../common/geolib.js"
+import { matchaMock, matchaMockMandataire, matchasMock } from "../mocks/matchas-mock.js"
+import { getOffreAvecInfoMandataire, getJobsFromElasticSearch } from "./formulaire.service.js"
+import { getApplicationByJobCount, IApplicationCount } from "./application.service.js"
+import { ILbaItem, LbaItem } from "./lbaitem.shared.service.types.js"
+import { IRecruiter } from "../common/model/schema/recruiter/recruiter.types.js"
+import { ILbaJobEsResult } from "./lbajob.service.types.js"
+
+/**
+ * Retourne les offres LBA correspondantes aux critères de recherche
+ * @param {string} romes une liste de codes romes séparée par des virgules
+ * @param {number} radius optionnel: le rayon de recherche pour une recherche centrée sur un point géographique
+ * @param {string} latitude optionnel: la latitude du centre de recherche
+ * @param {string} longitude optionnel: la longitude du centre de recherche
+ * @param {string} api le nom de l'api à des fins de traçage des actions
+ * @param {string} opco optionnel: le nom d'un opco pour ne retourner que les offres des sociétés affiliées à cet opco
+ * @param {string} opcoUrl optionnel: l'url d'un opco pour ne retourner que les offres des sociétés affiliées à cet opcoUrl
+ * @param {string} diploma optionnel: un fitre pour ne remonter ques les offres aboutissant à l'obtention du diplôme
+ * @param {string} caller optionnel: l'identifiant de l'utilisateur de l'api
+ * @param {string} useMock optionnel: un flag indiquant s'il faut retourner une valeur réelle ou une valeur mockée
+ * @returns {Promise<IApiError | { results: ILbaItem[] }>}
+ */
+export const getLbaJobs = async ({
+  romes,
+  radius,
+  latitude,
+  longitude,
+  api,
+  opco,
+  opcoUrl,
+  diploma,
+  caller,
+  useMock,
+}: {
+  romes: string
+  radius?: number
+  latitude?: string
+  longitude?: string
+  api: string
+  opco?: string
+  opcoUrl?: string
+  diploma?: string
+  caller?: string
+  useMock?: string
+}): Promise<IApiError | { results: ILbaItem[] }> => {
+  try {
+    const hasLocation = latitude === "" || latitude === undefined ? false : true
+
+    const distance = hasLocation ? radius || 10 : 21000
+
+    const params: any = {
```

Comment:
```suggestion
    const params = {
```
ne jamais utiliser any, sauf si on n'a pas le choix et c'est rarement le cas